packages/contracts-core/contracts/test/GovernanceMessage.t.sol

```diff
@@ -0,0 +1,488 @@
+    GovernanceMessage.Call[] diffCalls;
+
+    function test_getBatchHashFuzzedDifferentCalls(
+        bytes[255] memory data,
+        bytes32[255] memory to
+    ) public {
+        // types(uint8).max = 255
+        bytes memory serializedCall;
+        bytes memory prefix = abi.encodePacked(uint8(255));
+        bytes memory batch = prefix;
+        for (uint256 i; i < data.length; i++) {
+            diffCalls.push(GovernanceMessage.Call(to[i], data[i]));
+            serializedCall = abi.encodePacked(
+                to[i],
+                uint32(data[i].length),
+                data[i]
+            );
+            batch = abi.encodePacked(batch, serializedCall);
+        }
+        bytes32 batchHash = keccak256(batch);
+        assertEq(GovernanceMessage.getBatchHash(diffCalls), batchHash);
+    }
+
+    function test_isValidBatchDetectBatch() public pure {
+        // batch type in the form of a uint8
+        bytes memory data = hex"01";
+        // Append an empty bytes array of 32 bytes
+        data = abi.encodePacked(data, new bytes(32));
+        assert(GovernanceMessage.isValidBatch(data.ref(0)));
+    }
+
+    function test_isValidBatchDetectBatchFuzzed(bytes memory data) public {
+        vm.assume(data.length == 32);
+        data = abi.encodePacked(hex"01", data);
+        assert(GovernanceMessage.isValidBatch(data.ref(0)));
+    }
+
+    function test_isValidBatchWrongIdentifier() public {
+        // batch type in the form of a uint8
+        bytes memory data = hex"02";
+        // Append an empty bytes array of 32 bytes
+        data = abi.encodePacked(data, new bytes(32));
+        assertFalse(GovernanceMessage.isValidBatch(data.ref(0)));
+    }
+
+    function test_isValidBatchWrongIdentifierFuzzed(uint8 viewType) public {
+        // batch type in the form of a uint8
+        // Append an empty bytes array of 32 bytes
+        bytes memory data = abi.encodePacked(
+            abi.encodePacked(viewType),
+            new bytes(32)
+        );
+        if (viewType == 1) {
+            assert(GovernanceMessage.isValidBatch(data.ref(0)));
+        } else {
+            assertFalse(GovernanceMessage.isValidBatch(data.ref(0)));
+        }
+    }
+
+    function test_isValidBatchWrongIdentifierFuzzed(
+        uint8 viewType,
+        bytes memory data
+    ) public {
+        vm.assume(data.length == 32 && viewType != 1);
+        data = abi.encodePacked(abi.encodePacked(viewType), data);
+        assertFalse(GovernanceMessage.isValidBatch(data.ref(0)));
+    }
+
+    function test_isValidBatchWrongLength() public {
+        // batch type in the form of a uint8
+        bytes memory data = hex"01";
+        // Append an empty bytes array of 32 bytes
+        data = abi.encodePacked(data, new bytes(23));
+        assertFalse(GovernanceMessage.isValidBatch(data.ref(0)));
+    }
+
+    function test_isValidBatchWrongLengthFuzzed(bytes memory data) public {
+        vm.assume(data.length != 32);
```

Comment:
same question as above. instead of discarding 97% of tests, use a bytes32 as input